mapping:
    # for loop closure
    loop_keyframe_num_thld: 50
    loop_closure_search_radius: 10
    keyframe_search_num: 20
    loop_closure_fitness_score_thld: 0.1
    icp_downsamp_size: 0.1
    manually_loop_vaild_period: [99999999, 99999999]
    odom_loop_vaild_period: []
    scancontext_loop_vaild_period: []

    # imu -> gnss
    extrinsicT_imu2gnss: [ 0, 0, 0 ]
    extrinsicR_imu2gnss: [ 1, 0, 0, 
                           0, 1, 0, 
                           0, 0, 1 ]

official:
    save_globalmap_en: false
    save_resolution: 0.2
    prior_map_path: "/home/will/data/test_mapping/mapping1"
    stitch_map_path: "/home/will/data/test_mapping/mapping2"
    result_map_path: "/home/will/data/test_mapping/mapping3"

# ============relocalization_cfg============
relocalization_cfg:
    algorithm_type: "scan_context"
    # algorithm_type: "manually_set"

utm_origin:
    zone: "51N"
    east: 536808.26993
    north: 4501498.61740
    up: 44.21500

scan_context:
    lidar_height: 2.183
    sc_dist_thres: 0.7

bnb3d:
    linear_xy_window_size: 5   # meter
    linear_z_window_size: 0.5  # meter
    angular_search_window: 180  # degree 0~180
    pc_resolutions: [0.2, 0.3, 0.5]
    bnb_depth: 3
    min_score: 0.3
    enough_score: 0.7

    min_xy_resolution: 0.2
    min_z_resolution: 0.1
    min_angular_resolution: 5   # degree

    filter_size_scan: 0.2
    debug_mode: false

ndt:
    step_size: 0.1
    resolution: 1

gicp:
    use_gicp: false
    filter_range: 80.0
    gicp_downsample: 0.1
    search_radius: 0.5
    teps: 1.0e-4
    feps: 0.001
    fitness_score: 0.3
